app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -131,5 +139,10 @@ def find_default_organization_and_or_environment
         fail HttpErrors::NotFound, _("You have not set a default organization on the user %s.") % current_user.login
       end
     end
+
+    def find_host_with_subscriptions(id, permission)
+      @host = resource_finder(::Host.authorized(permission), id)
+      fail _("Host has not been registered with subscription-manager") if @host.subscription_facet.nil?
```

Comment:
I think this will result in a 500 instead of a more sane response code for a user?